### ISSUE TITLE
Select2 view does not update when model is changed

### ISSUE THREAD

**Issue by JeremyFriesen** [angular-ui/ui-select2#261]
I'll break my problem down, and hopefully someone here can offer a solution.

The Requirements:
1. I need a dropdown that is compatible with AngularJS, and Bootstrap
- Select2 looks like a good balance between completeness of functionality and compatible with the above.
  1. I need a drop-down that can be configured to accept a directive like ng-options.  Select2 (out of the box) can accept ng-options only if applied to a &lt;select&gt; tag I need to use the input tag.
  2. I need a drop-down that can be configured to only select one option (Select2 can by not using the 'multiple' directive)
  3. I need a drop-down that I can add an additional directive 'customModel' which will allow me to save typed in text to the drop-down in the place defined in 'custom-model'.

The Example (this is what I would like to end up with).
  <input type="text" ui-select2 ng-model="SelectedItemId" input-options="item.Id as item.Name for item in Items" custom-model="CustomItemName" />

The Explanation:
- ng-model (SelectedItemId) will be null if custom text is entered, otherwise it will be the selected item's Id value.
- custom-model (CustomItemName) will be the custom text if entered, otherwise, null if an item from $scope.Items is selected.
- input-options will work just as ng-options works on a plain select.

The Code:
- I have written directives for input-options and custom-model (and though they are not completed, I'm having some success with them (see plunk: http://plnkr.co/edit/wQSJ4UKAlXtui2NZ84hR?p=info)

The Plunk:
- http://plnkr.co/edit/wQSJ4UKAlXtui2NZ84hR?p=preview

The Problem:
I would expect the Select2's view to change when clicking on the 'increment' button on the "Problem 2" example, but it doesn't.

**Comment by dulizhao** [comment]
Has this issue been fixed?  I met same problem. I did not know how to change status selected use of javascript.

**Comment by JeremyFriesen** [comment]
No, I'm now using ui-select in angular-ui.  https://github.com/angular-ui/ui-select

And found a work-around in my project to avoid the need for this feature.  

Good luck!
